fem/coefficient.hpp

```diff
@@ -787,10 +802,25 @@ class MatrixFunctionCoefficient : public MatrixCoefficient
       mat.SetSize(0);
    }
 
+   /// Construct a symmetric square matrix coefficient from a C-function
```

Comment:
Needs a reference to `MatrixCoefficient::EvalSymmetric` or something similar to define how the vector function will be interpreted as a symmetric matrix